### PR TITLE
feat: add resource pools

### PR DESCRIPTION
**Pull request by armandmcqueen**
## Description

Merge in resource pools feature branch. Convert the internal configuration from being a hybrid (where both pre-resourcepool fields exist and resource pool-centric configuration exists) to only being resource-pool-centric. 

Add in backwards compatibility that converts legacy master.yaml to new master.yaml format when parsing config file. 

Update det-deploy to deploy with multiple resource pools by default.

Update documentation to include information about resource pools.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by armandmcqueen** [comment]
@shiyuann I think I rebased this correctly, but I'm not 100% certain so it would be great if you could look through it and confirm.

**Comment by shiyuann** [comment]
@armandmcqueen Do you want me to review it now or is it still WIP?

**Comment by armandmcqueen** [comment]
@shiyuann  It would be great if you could review it now to make sure the rebase didn't accidentally change any behavior

File: docs/reference/cluster-config.txt [hunk 1]
```diff
@@ -379,256 +305,387 @@ The master supports the following configuration settings:
...
+   -  ``description``: The description of the resource pool.
+
+   -  ``max_cpu_containers_per_agent``: The maximum number of containers
+      that do not take any GPUs on each agent.
+
+   -  ``scheduler``: Specifies how Determined schedules tasks to agents.
+      The scheduler configuration on each resource pool will override
+      the global one.
+
+      -  ``type``: The scheduling policy to use when allocating
```
Comment (shiyuann) [inline docs/reference/cluster-config.txt:410]: There should be three types, `fair_share`, `priority`, and `round-robin`. The priority scheduler configuration should probably use the same fields as the global scheduler configuration.

**Review by shiyuann — APPROVED** [review]
LGTM!

**Pull request merged**